lib/dataset/test/histogram_test.cpp

```diff
@@ -311,6 +336,34 @@ TEST_F(Histogram1DTest, coord_name_differs_dim) {
                                  Values{19, 12, 12}));
 }
 
+TEST_F(Histogram1DTest, int64_weights) {
```

Comment:
Also add a test for `int32`?

Reply:
What about the other 10+ cases? What would we gain?

See also discussion in #3251.